mteb/abstasks/AbsTaskInstructionRetrieval.py

```diff
@@ -0,0 +1,611 @@
+import json
+import logging
+import os
+from collections import defaultdict
+from time import time
+from typing import Dict, List, Tuple
+
+import tqdm
+from datasets import Features, Value, load_dataset
+
+from ..evaluation.evaluators import utils
+from ..evaluation.evaluators.InstructionRetrievalEvaluator import InstructionRetrievalEvaluator
+from .AbsTask import AbsTask
+from .AbsTaskRetrieval import HFDataLoader, AbsTaskRetrieval
+
+logger = logging.getLogger(__name__)
+
+
+# Adapted from https://github.com/beir-cellar/beir/blob/f062f038c4bfd19a8ca942a9910b1e0d218759d4/beir/datasets/data_loader_hf.py#L10
+class HFDataLoaderInstructions(HFDataLoader):
```

Comment:
I kept this as similar to the HFDataLoader as possible and inherit some methods from it, like `load_corpus`. Others had to be different because of the instructions portion.